### PR TITLE
Add check of availability of VM role size in location

### PR DESCRIPTION
**Pull request by jen20**
There was previously nothing to prevent creation of a virtual machine of an instance size which was unavailable in a given location - instead the operation completed and getting the status returned a `ResourceNotFound` error. This pull request makes the following changes:

- Add a `GetLocation` function to the `locationClient` in order to get access to the structure without iterating through the list returned from `GetLocationList` in multiple spots.
- Remove the call to verify the instance size against the global list.
- Add a function `verifyInstanceSizeAvailableInLocation` to the `vmClient` and call this in place of verification against the global list.

### CODE REVIEW

File: clients/locationClient/locationClient.go [hunk 1]
```diff
@@ -55,3 +55,29 @@ func GetLocationList() (LocationList, error) {
 
 	return locationList, nil
 }
+
+func GetLocation(location string) (*Location, error) {
+	if len(location) == 0 {
+		return nil, fmt.Errorf(azure.ParamNotSpecifiedError, "location")
```
Comment (ahmetalpbalkan) [inline clients/locationClient/locationClient.go]: if we're always formatting this azure.ParamNotSpecifiedError string, it shouldn't probably be exposed. can you please change while you're at it, or maybe in a separate PR?

File: clients/locationClient/locationClient.go [hunk 2]
```diff
@@ -55,3 +55,29 @@ func GetLocationList() (LocationList, error) {
...
+
+		return &existingLocation, nil
+	}
+
+	var availableLocations bytes.Buffer
+	for _, existingLocation := range locations.Locations {
+		availableLocations.WriteString(existingLocation.Name + ", ")
+	}
+
+	return nil, errors.New(fmt.Sprintf(invalidLocationError, location, strings.Trim(availableLocations.String(), ", ")))
```
Comment (ahmetalpbalkan) [inline clients/locationClient/locationClient.go]: having a Stringer on Location and just using string.Join(",", locations.Locations) might be a neat solution here.

Comment (jen20) [inline clients/locationClient/locationClient.go]: Agreed - I just took the code from the existing functions in the `locationsClient` here. Will audit the entire code base for these and fix them on a separate pull request.

File: clients/locationClient/locationClient.go (hunk 1 above)
Comment (jen20) [inline clients/locationClient/locationClient.go]: :+1: will do on a separate PR

File: clients/vmClient/vmClient.go [hunk 3]
```diff
@@ -856,4 +857,23 @@ next:
 	return nil
 }
 
+func verifyInstanceSizeAvailableInLocation(location *locationClient.Location, vmSize string) error {
```
Comment (ahmetalpbalkan) [inline clients/vmClient/vmClient.go]: Verify-> is. And this should return (bool, err) and the caller of the method should craft the error. This will allow this method to be reused in different places.

Comment (jen20) [inline clients/vmClient/vmClient.go]: I've done this, however it means either an unnecessary API call or removing the list of available sizes from the error message. In this instance I've chosen to remove the list of available sizes.

File: clients/locationClient/locationClient.go (hunk 2 above)
Comment (hopetobelievein) [inline clients/locationClient/locationClient.go]: And can you move this part of the code(I mean converting available locations to a string and creating an error) to separate method to avoid code duplication? It then can be used in both GetLocation and ResolveLocation.

Comment (jen20) [inline clients/locationClient/locationClient.go]: Will put in a separate PR at the same time as replacing the buffer/writer with a string join.

File: clients/locationClient/locationClient.go (hunk 1 above)
Comment (jen20) [inline clients/locationClient/locationClient.go]: We have a few options on this - it's used throughout the codebase hence the original decision (I assume) to export it. We could either duplicate wherever necessary and make it internal or leave it as is. Any preference?

Comment (jen20) [inline clients/locationClient/locationClient.go]: See #28 for an implementation of this.

File: clients/locationClient/locationClient.go (hunk 2 above)
Comment (jen20) [inline clients/locationClient/locationClient.go]: See #29 for an implementation of `Stringer` on `LocationList`.
